src/engine/Scene.ts

```diff
@@ -307,43 +307,32 @@ export class Scene extends Class implements ICanInitialize, ICanActivate, ICanDe
     */
    public update(engine: Engine, delta: number) {
       this._preupdate(engine, delta);
-      var i: number, len: number;
-
-      
 
       // Remove timers in the cancel queue before updating them
-      for (i = 0, len = this._cancelQueue.length; i < len; i++) {
-         this.removeTimer(this._cancelQueue[i]);
-      }
+      this._cancelQueue.forEach((timer) => this.removeTimer(timer));
+
       this._cancelQueue.length = 0;
 
       // Cycle through timers updating timers
-      for (var timer of this._timers) {
-         timer.update(delta);
-      };
+      this._timers.forEach((timer) => timer.update(delta));
 
       // Cycle through actors updating UI actors
-      for (i = 0, len = this.uiActors.length; i < len; i++) {
-         this.uiActors[i].update(engine, delta);
-      }
+      this.uiActors.forEach((uiActor) => uiActor.update(engine, delta));
 
       // Cycle through actors updating tile maps
-      for (i = 0, len = this.tileMaps.length; i < len; i++) {
-         this.tileMaps[i].update(engine, delta);
-      }
+      this.tileMaps.forEach((tileMap) => tileMap.update(engine, delta));
 
       // Cycle through actors updating actors
-      for (i = 0, len = this.actors.length; i < len; i++) {
-         this.actors[i].update(engine, delta);
-      }
+      this.actors.forEach((actor) => actor.update(engine, delta));
 
       // Cycle through triggers updating
-      for (i = 0, len = this.triggers.length; i < len; i++) {
-         this.triggers[i].update(engine, delta);
-      }
+      this.triggers.forEach((trigger) => trigger.update(engine, delta));
 
       this._collectActorStats(engine);
 
+      // propagates all events through their paths assigned
+      engine.input.pointers.propagate();
```

Comment:
Is this going to cause a performance hit by running propagation every frame? Pointer events won't necessarily fire every frame.

Can we move propagation to be contained in the `CapturePointer.ts` trait where it only runs if an actor has opted in to pointer events?

Reply:
The thing is, that events are propagated by paths assigned to them. If Event's path is empty, nothing'll happen. I moved assignment of event's paths to CapturePointer Trait. So event'll be updated not every frame, but only when the happen.